airflow_dbt_python/hooks/target.py

```diff
@@ -0,0 +1,351 @@
+        """Returns the override_name if defined, otherwise defaults to name.
+
+        >>> DbtConnectionParam("login", "user").override_name
+        'user'
+        >>> DbtConnectionParam("port").override_name
+        'port'
+        """
+        if self.store_override_name is None:
+            return self.name
+        return self.store_override_name
+
+
+class DbtConnectionHookMeta(ABCMeta):
+    """A hook metaclass to collect all subclasses of DbtConnectionHook."""
+
+    _dbt_hooks_by_conn_type: ClassVar[dict[str, DbtConnectionHookMeta]] = {}
+    conn_type: str
+
+    def __new__(cls, name, bases, attrs, **kwargs) -> DbtConnectionHookMeta:
+        """Adds each DbtConnectionHook subclass to the dict based on its conn_type."""
+        new_hook_cls = super().__new__(cls, name, bases, attrs)
+        if new_hook_cls.conn_type in cls._dbt_hooks_by_conn_type:
+            warnings.warn(
+                f"The connection type `{new_hook_cls.conn_type}`"
+                f" has been overwritten by `{new_hook_cls}`",
+                UserWarning,
+                stacklevel=1,
+            )
+
+        cls._dbt_hooks_by_conn_type[new_hook_cls.conn_type] = new_hook_cls
+        return new_hook_cls
+
+
+class DbtConnectionHook(BaseHook, ABC, metaclass=DbtConnectionHookMeta):
+    """A hook to get a dbt profile based on the Airflow connection."""
+
+    conn_type = "dbt"
+    hook_name = "dbt Hook"
+
+    conn_params: list[Union[DbtConnectionParam, str]] = [
+        DbtConnectionParam("conn_type", "type"),
+        "host",
+        "schema",
+        "login",
+        "password",
+        "port",
+    ]
+    conn_extra_params: list[Union[DbtConnectionParam, str]] = []
+
+    def __init__(
+        self,
+        *args,
+        conn: Connection,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self.conn = conn
+
+    @classmethod
+    def get_db_conn_hook(cls, conn_id: str) -> DbtConnectionHook:
+        """Get a dbt hook class depend on Airflow connection type."""
+        conn = cls.get_connection(conn_id)
+
+        if hook_cls := cls._dbt_hooks_by_conn_type.get(conn.conn_type):
+            return hook_cls(conn=conn)
+        raise KeyError(
+            f"There are no DbtConnectionHook subclasses with conn_type={conn.conn_type}"
+        )
+
+    def get_dbt_target_from_connection(self) -> Optional[dict[str, Any]]:
+        """Return a dictionary of connection details to use as a dbt target.
+
+        The connection details are fetched from an Airflow connection identified by
+        self.dbt_conn_id.
+
+        Returns:
+            A dictionary with a configuration for a dbt target, or None if a matching
+                Airflow connection is not found for given dbt target.
+        """
+        details = self.get_dbt_details_from_connection(self.conn)
+
+        return {self.conn.conn_id: details}
+
+    def get_dbt_details_from_connection(self, conn: Connection) -> dict[str, Any]:
```

Comment:
Okay, this is no longer a classmethod, safe to ignore my comment on this topic. I think this is now fine.